e2e/testcases/reconciler_manager_test.go

```diff
@@ -1116,3 +1120,91 @@ func filterResourceMap(resourceMap map[string]v1beta1.ContainerResourcesSpec, co
 	}
 	return filteredMap
 }
+
+// TestReconcilerManagerRSyncCRDMissing validates that the reconciler-manager
+// stays up when the RootSync CRD is deleted.
+func TestReconcilerManagerRootSyncCRDMissing(t *testing.T) {
+	nt := nomostest.New(t, nomostesting.ACMController,
+		ntopts.WithDelegatedControl, ntopts.Unstructured)
+
+	reconcilerManagerKey := client.ObjectKey{
+		Name:      reconcilermanager.ManagerName,
+		Namespace: configsync.ControllerNamespace,
+	}
+
+	t.Log("Validate the reconciler-manager deployment is healthy")
+	nt.Must(nt.Validate(reconcilerManagerKey.Name, reconcilerManagerKey.Namespace, &appsv1.Deployment{},
+		testpredicates.StatusEquals(nt.Scheme, kstatus.CurrentStatus)))
+
+	nt.Must(nomostest.UninstallRootSyncCRD(nt))
+
+	t.Log("Validate the reconciler-manager deployment stays healthy for 30s")
+	err := nt.Watcher.WatchObject(kinds.Deployment(),
+		reconcilerManagerKey.Name, reconcilerManagerKey.Namespace,
+		[]testpredicates.Predicate{
+			testpredicates.StatusNotEquals(nt.Scheme, kstatus.CurrentStatus),
+		}, testwatcher.WatchTimeout(30*time.Second))
+	require.Error(t, err)
+	if !errors.Is(err, context.DeadlineExceeded) {
+		nt.T.Fatalf("expected watch to time out, but got unexpected error: %v", err)
+	}
+
+	t.Log("Deleting the reconciler-manager deployment pod")
+	nt.Must(nt.KubeClient.DeleteAllOf(&corev1.Pod{},
+		client.InNamespace(configsync.ControllerNamespace),
+		client.MatchingLabels{metadata.DeploymentNameLabel: reconcilermanager.ManagerName}))
+
+	t.Log("Waiting for the reconciler-manager deployment to become unhealthy")
+	nt.Must(nt.Watcher.WatchObject(kinds.Deployment(),
+		reconcilerManagerKey.Name, reconcilerManagerKey.Namespace,
+		[]testpredicates.Predicate{
+			testpredicates.StatusNotEquals(nt.Scheme, kstatus.CurrentStatus),
+		}))
+
+	t.Log("Waiting for the reconciler-manager deployment pod to become healthy")
+	// Can't use the Watcher on a List (yet), so poll instead.
+	nomostest.Wait(nt.T, "wait for reconciler-manager deployment pod to become healthy", nt.DefaultWaitTimeout, func() error {
+		return validateReconcilerManagerPodStatusCurrent(nt)
+	})
+
+	t.Log("Waiting for the reconciler-manager deployment to become healthy")
+	nt.Must(nt.Watcher.WatchForCurrentStatus(kinds.Deployment(),
+		reconcilerManagerKey.Name, reconcilerManagerKey.Namespace))
+
+	nt.Must(nomostest.InstallRootSyncCRD(nt))
+
+	t.Log("Validate the reconciler-manager deployment stays healthy for 30s")
```

Comment:
As we discussed offline, this assertion could be improved by instead creating a `RootSync` and ensuring that it reconciles

Reply:
re-wrote the test here to validate RootSync and RepoSync spec updates and syncing as proof of reconciler-manager health.